packages/sdk/tests/e2e/dataProtectorSharing/consumeProtectedData.test.ts

```diff
@@ -1,159 +1,277 @@
+    await dataProtectorCreator.sharing.setProtectedDataToSubscription({
+      protectedData,
+    });
+    const rentingParams = {
+      price: 0,
+      duration: 30 * 24 * 60 * 60,
+    };
+    await dataProtectorCreator.sharing.setProtectedDataToRenting({
+      protectedData,
+      ...rentingParams,
+    });
 
-        await dataProtectorConsumer.sharing.subscribeToCollection({
-          collectionId,
-          ...subscriptionParams,
-        });
+    await dataProtectorConsumer.sharing.subscribeToCollection({
+      collectionId,
+      ...subscriptionParams,
+    });
+  }, timeouts.createAddOnlyAppWhitelist + timeouts.createAndPublishWorkerpoolOrder + timeouts.protectData + timeouts.createCollection + timeouts.addToCollection + timeouts.setSubscriptionParams + timeouts.setProtectedDataToSubscription + timeouts.setProtectedDataToRenting + timeouts.subscribe);
 
-        // --- WHEN
-        const onStatusUpdateMock = jest.fn();
-        const result = await dataProtectorConsumer.sharing.consumeProtectedData(
-          {
+  describe('When whitelist contract does not have registered delivery app', () => {
+    it(
+      'should throw error',
+      async () => {
+        await expect(
+          dataProtectorConsumer.sharing.consumeProtectedData({
             app: DEFAULT_PROTECTED_DATA_DELIVERY_APP,
             protectedData,
             workerpool: WORKERPOOL_ADDRESS,
-            onStatusUpdate: onStatusUpdateMock,
-          }
+            maxPrice: 1000,
+          })
+        ).rejects.toThrow(
+          `This whitelist contract does not have registered this app: ${DEFAULT_PROTECTED_DATA_DELIVERY_APP.toLocaleLowerCase()}.`
         );
+      },
+      timeouts.consumeProtectedData
+    );
+  });
+
+  describe('When whitelist contract registered delivery app', () => {
+    beforeAll(async () => {
+      // add app 'protected-data-delivery-dapp.apps.iexec.eth' to AppWhitelist SC
+      await dataProtectorCreator.sharing.addAppToAddOnlyAppWhitelist({
+        addOnlyAppWhitelist,
+        app: DEFAULT_PROTECTED_DATA_DELIVERY_APP,
+      });
+    }, timeouts.addAppToAddOnlyAppWhitelist);
 
-        // --- THEN
-        expect(onStatusUpdateMock).toHaveBeenCalledWith({
-          title: 'CONSUME_RESULT_COMPLETE',
-          isDone: true,
+    it(
+      'should create a deal with valid inputs',
+      async () => {
+        await depositNRlcForAccount(walletConsumer.address, 1000_000);
+        await approveAccount(
+          walletConsumer.privateKey,
+          DEFAULT_SHARING_CONTRACT_ADDRESS,
+          1000_000
+        );
+        let testResolve;
+        const testPromise = new Promise((resolve) => {
+          testResolve = resolve;
         });
-        console.log(result);
+
+        const status = [];
+        const onStatusUpdate = ({ title, isDone, payload }) => {
+          status.push({ title, isDone, payload });
+          if (title === 'CONSUME_TASK') {
+            testResolve();
+          }
+        };
```

Comment:
It works i agree but it's not very clean. I think we should try to use mock instead but it will take more time (ask pierre => voucher lead if it's ok or not 🤷🏻‍♂️)